includes/Core/McpClient.php

```diff
@@ -0,0 +1,541 @@
+<?php
+/**
+ * MCP Client for connecting to external MCP servers.
+ *
+ * @package McpAdapter
+ */
+
+declare( strict_types=1 );
+
+namespace WP\MCP\Core;
+
+use WP\MCP\Infrastructure\ErrorHandling\Contracts\McpErrorHandlerInterface;
+use WP\MCP\Infrastructure\Observability\Contracts\McpObservabilityHandlerInterface;
+
+/**
+ * MCP Client for connecting to external MCP servers.
+ * 
+ * Mirrors the McpServer architecture for consistency.
+ */
+class McpClient {
+
+	/**
+	 * Client identifier.
+	 *
+	 * @var string
+	 */
+	private string $client_id;
+
+	/**
+	 * Server URL.
+	 *
+	 * @var string
+	 */
+	private string $server_url;
+
+	/**
+	 * Client configuration.
+	 *
+	 * @var array
+	 */
+	private array $config;
+
+	/**
+	 * Error handler.
+	 *
+	 * @var McpErrorHandlerInterface
+	 */
+	public McpErrorHandlerInterface $error_handler;
+
+	/**
+	 * Observability handler.
+	 *
+	 * @var McpObservabilityHandlerInterface
+	 */
+	public McpObservabilityHandlerInterface $observability_handler;
+
+	/**
+	 * Connection status.
+	 *
+	 * @var bool
+	 */
+	private bool $connected = false;
+
+	/**
+	 * Discovered server capabilities.
+	 *
+	 * @var array
+	 */
+	private array $capabilities = array();
+
+	/**
+	 * MCP session ID for DeepWiki and similar servers.
+	 *
+	 * @var string|null
+	 */
+	private ?string $session_id = null;
+
+	/**
+	 * Transport protocol (sse or mcp).
+	 *
+	 * @var string
+	 */
+	private string $transport = 'mcp';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string                           $client_id             Client identifier.
+	 * @param string                           $server_url            MCP server URL.
+	 * @param array                            $config                Client configuration.
+	 * @param McpErrorHandlerInterface         $error_handler         Error handler.
+	 * @param McpObservabilityHandlerInterface $observability_handler Observability handler.
+	 */
+	public function __construct(
+		string $client_id,
+		string $server_url,
+		array $config,
+		McpErrorHandlerInterface $error_handler,
+		McpObservabilityHandlerInterface $observability_handler
+	) {
+		$this->client_id             = $client_id;
+		$this->server_url            = rtrim( $server_url, '/' );
+		$this->config                = $config;
+		$this->error_handler         = $error_handler;
+		$this->observability_handler = $observability_handler;
+
+		// Detect transport protocol from URL
+		if ( strpos( $server_url, '/sse' ) !== false ) {
+			$this->transport = 'sse';
+		} elseif ( strpos( $server_url, '/mcp' ) !== false ) {
+			$this->transport = 'mcp';
+		}
+
+
+		// Auto-connect on construction
+		$this->connect();
+	}
+
+	/**
+	 * Connect to the MCP server.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	public function connect(): bool {
```

Comment:
Is this meant to be consumed publicly? e.g. ` if ( ! $my_client->is_connected() ) { $client->connect(); }`

If not, let's make the method `private` until we have a concrete use case to expose it `public`.

If so, I suggest we get it out of constructor and dogfood our own APIs by calling it in `MCPAdapter::create_client()`.